src/Conf.ml

```diff
@@ -916,10 +963,15 @@ let update_using_xdg =
   | None -> Staged.stage (fun conf -> conf)
 
 let build_config ~filename =
-  default
-  |> Staged.unstage update_using_xdg
-  |> read_config ~filename ~parents:true
-  |> update_using_env |> C.update_using_cmdline
+  match !profile with
+  | Some profile ->
+      let f (_, x, _, p) = if Poly.(x = profile) then Some p else None in
+      List.find_map_exn preset_profiles ~f
```

Comment:
I'm don't think we want to disallow customisation here.
I would also expect a "default" preset, similar to ocp-indent.

Reply:
To my understanding, we may encounter a profile setting via any of the existing ways of configuration. Semantically, processing a profile option should set all the config options according to the profile, overriding everything in any lower priority config settings. And then processing of the rest of the config options should proceed, enabling customization of a profile one option at a time.

Reply:
For cases where there are profile and other options on the command line, I think it makes sense to take the profile as lower priority than the other options.